lib/api/core/models/repositories/profileRepository.js

```diff
@@ -176,13 +178,23 @@ module.exports = function (kuzzle) {
       return q.reject(new BadRequestError(profile._id + ' is one of the basic profiles of Kuzzle, you cannot delete it, but you can edit it.'));
     }
 
-    return this.deleteFromDatabase(profile._id)
+    filter = { or: [{terms: { 'profiles': [ profile._id ] }}] };
+
+    return kuzzle.repositories.user.search(filter, 0, 1, false)
       .then(response => {
-        if (this.profiles[profile._id]) {
-          delete this.profiles[profile._id];
+        if (response.total > 0) {
+          return q.reject(new BadRequestError('The profile "' + profile._id + '" cannot be deleted since it is used by some users.'));
```

Comment:
I agree, a `ForbiddenError` seems better in that case, since:
- the request is well-formed (thus it's not a `BadRequest`)
- we forbid this action because other users use the profile